Content.Shared/Xenoarchaeology/Artifact/SharedXenoArtifactSystem.Unlock.cs

```diff
@@ -0,0 +1,95 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Content.Shared.Xenoarchaeology.Artifact.Components;
+
+namespace Content.Shared.Xenoarchaeology.Artifact;
+
+public abstract partial class SharedXenoArtifactSystem
+{
+    private EntityQuery<XenoArtifactUnlockingComponent> _unlockingQuery;
+
+    // TODO: we should cancel the unlock state if you remove/add nodes
```

Comment:
does that mean that on add/remove node we should re-shuffle? or added node should be locked? thats expected but todo is awfully specific... what about remove?...

Reply:
so after inspecting the code - it seems that 'holes' inside artifact grid are kinda 'pre-determined' and there is no need to re-shuffle nodes as of now.